subcommands/keys/tuf_utils.go

```diff
@@ -361,15 +363,27 @@ func genProdTufRoot(ciRoot *client.AtsTufRoot) (prodRoot *client.AtsTufRoot) {
 }
 
 func signNewTufRoot(curCiRoot, newCiRoot, newProdRoot *client.AtsTufRoot, creds OfflineCreds) {
-	// Always sign with new root key; sign with old root key if it was rotated.
-	oldKey, err := FindOneTufSigner(curCiRoot, creds, curCiRoot.Signed.Roles["root"].KeyIDs)
-	subcommands.DieNotNil(err)
-	newKey, err := FindOneTufSigner(newCiRoot, creds, newCiRoot.Signed.Roles["root"].KeyIDs)
-	subcommands.DieNotNil(err)
-	signers := []TufSigner{newKey}
-	if oldKey.Id != newKey.Id {
-		signers = append(signers, oldKey)
+	// Find new and old keys that match; at least one of them must be found.
+	oldKey, oldErr := FindOneTufSigner(curCiRoot, creds, curCiRoot.Signed.Roles["root"].KeyIDs)
+	newKey, newErr := FindOneTufSigner(newCiRoot, creds, newCiRoot.Signed.Roles["root"].KeyIDs)
```

Comment:
if `newErr` is not `errFoundNoKey` we can `subcommands.DieNotNil(newErr)` at this point?

Reply:
I played with this code quite a bit.
What I have is the minimal and clear enough variant.

Initially, I was doing it like you suggest, and that ended up into too many "if" statements, making it hard to follow.

Reply:
ok, that's what I thought.

Reply:
Actually, I tried to reshuffle it in a bit more plausible way: https://github.com/foundriesio/fioctl/pull/240/commits/ff987dce2edb913341220d3a7e617f4cdaeea2fe

Please, let me know if that looks better to you.